drivers/ethernet/eth_stellaris.c

```diff
@@ -0,0 +1,366 @@
+				    (u8_t *)&reg_val, K_NO_WAIT)) {
+			return -1;
+		}
+		bytes_left -= 4;
+	}
+	return frame_len;
+}
+
+static void eth_stellaris_rx(struct device *dev)
+{
+	struct net_pkt *pkt = NULL;
+	struct eth_stellaris_runtime *dev_data = DEV_DATA(dev);
+	struct net_if *iface = dev_data->iface;
+	struct net_eth_hdr *pkt_hdr;
+	int frame_len, ret;
+
+	/* Obtain the packet to be populated */
+	pkt = net_pkt_get_reserve_rx(0, K_NO_WAIT);
+	if (!pkt) {
+		SYS_LOG_ERR("Could not allocate pkt");
+		goto err_mem;
+	}
+
+	frame_len = eth_stellaris_rx_pkt(dev, pkt);
+	if (frame_len == -1) {
+		goto err_rx;
+	}
+
+	ret = net_recv_data(iface, pkt);
+	if (ret < 0) {
+		SYS_LOG_ERR("Failed to place frame in RX Queue");
+		goto pkt_unref;
+	}
+
+	/* Update statistics counters */
+	eth_stats_update_bytes_rx(iface, frame_len - 6);
+	eth_stats_update_pkts_rx(iface);
+	pkt_hdr = NET_ETH_HDR(pkt);
+	if (net_eth_is_addr_broadcast(&pkt_hdr->dst)) {
+		eth_stats_update_broadcast_rx(iface);
+	} else if (net_eth_is_addr_multicast(&pkt_hdr->dst)) {
+		eth_stats_update_multicast_rx(iface);
+	}
+	return;
+
+err_rx:
+	SYS_LOG_ERR("Failed to append data to buffer");
+
+pkt_unref:
+	net_pkt_unref(pkt);
+
+err_mem:
+	eth_stellaris_rx_error(iface);
+}
+
+static void rx_isr(void *arg)
+{
+	/* Read the interrupt status */
+	struct device *dev = (struct device *)arg;
+	struct eth_stellaris_runtime *dev_data = DEV_DATA(dev);
+	int isr_val = sys_read32(REG_MACRIS);
+	u32_t lock;
+
+	lock = irq_lock();
+
+	/* Acknowledge the interrupt */
+	sys_write32(isr_val, REG_MACRIS);
+
+	if (isr_val & BIT_MACRIS_RXINT) {
+
+		eth_stellaris_rx(dev);
+	}
+
+	if (isr_val & BIT_MACRIS_TXEMP) {
+
+		dev_data->tx_err = false;
+		k_sem_give(&dev_data->tx_sem);
+	}
+
+	if (isr_val & BIT_MACRIS_TXER) {
+
+		SYS_LOG_ERR("Txn Frame Error");
+		eth_stats_update_errors_tx(dev_data->iface);
+		dev_data->tx_err = true;
+		k_sem_give(&dev_data->tx_sem);
+	}
+
+	if (isr_val & BIT_MACRIS_RXER) {
+
+		SYS_LOG_ERR("Error Frame Recieved");
+		eth_stellaris_rx_error(dev_data->iface);
+	}
+	irq_unlock(lock);
+}
+
+static void eth_stellaris_init(struct net_if *iface)
+{
+	struct device *dev;
+	struct eth_stellaris_config *dev_conf;
+	struct eth_stellaris_runtime *dev_data;
+
+	dev = net_if_get_device(iface);
+	dev_data = DEV_DATA(dev);
+	dev_data->iface = iface;
+	dev_conf = DEV_CFG(dev);
+
+	/*1. Assign link local address */
+	net_if_set_link_addr(iface,
+			     dev_data->mac_addr, 6, NET_LINK_ETHERNET);
+
+	ethernet_init(iface);
+
+	/* Initialize semaphore */
+	k_sem_init(&dev_data->tx_sem, 0, 1);
+
+	/* Initialize Interrupts */
+	dev_conf->config_func(dev);
+}
+
+static struct net_stats_eth *eth_stellaris_stats(struct device *dev)
+{
+	return &(DEV_DATA(dev)->stats);
+}
+
+static void eth_stellaris_irq_config(struct device *dev);
+
+struct eth_stellaris_config eth_cfg = {
+	.mac_base = CONFIG_ETH_BASE_ADDR,
+	.config_func = eth_stellaris_irq_config,
+};
+
+struct eth_stellaris_runtime eth_data = {
+	.mac_addr = {
```

Comment:
You are allowing user to specify any MAC address here, should we limit this to bytes 3-6 as the first three should be TI specific?

Reply:
We are not very sure about this, there doesn't seem to be any specific OUI recommended by TI.

Reply:
We are using 00-12-4B in cc2520 and cc1200 drivers, I suppose that could be used here too. There are also other options mentioned in https://hwaddress.com/?q=texas%20instruments

Reply:
@jukkar Not sure why we would want to do that. The user should be free to assign any MAC address to the interface.

Reply:
User can certainly change anything as the code can be tweaked. I was just suggesting some sane defaults for the default mac address.

Reply:
The DTS has been updated to use the RFC 7042 MAC address that you suggested as the default.